### DISCUSSION TITLE
There should be an api for deleting folders

### DISCUSSION THREAD

**Discussion by renjiali** [supabase/supabase#33793] [Feature Requests]
I use this for my cloud storage and currently have a problem when deleting files.
I operate in this way, for example, if there is an activity, I put the activity under the bucket of "activity" with the activity id. The main image of the activity is directly placed in the folder named after the activity id and renamed as "activity". I want to group their images, for example, all other images are under "images", but currently it's very annoying when deleting them. I have to search through all files, but if it's possible to delete a folder. I can just delete the folder named after the activity id.

**Comment by encima** [comment]
You can pass the name of the folder to the `remove` method within a bucket and delete the folder this way.

Storage also supports the S3 protocol so you can delete folders using any S3 library or client

**Comment by j-mendez** [comment]
@encima that does not work to pass a folder in the remove method to delete it with the js client.

**Comment by encima** [comment]
@j-mendez ah, my bad, I was not aware of that, thanks!

I'll add the enhancement tag back to this issue then. Workaround for now would be to use an S3 client but I'll pass this to the storage team to comment

**Comment by arthurgavazza** [comment]
Is anyone actively working on this issue? If not, I’d love to contribute a PR. Since I'm new to this repo, any guidance would be greatly appreciated!

**Comment by Hallidayo** [comment]
Hi everyone, due to inactivity on this issue I've moved the issue over to discussions/enhancements.
